prov/verbs/src/verbs_info.c

```diff
@@ -1051,18 +1091,13 @@ int fi_ibv_init_info(const struct fi_info **all_infos)
 	}
 
 	for (i = 0; i < num_devices; i++) {
-		ret = fi_ibv_alloc_info(ctx_list[i], &fi, &verbs_msg_domain);
-		if (!ret) {
-			if (!*all_infos)
-				*all_infos = fi;
-			else
-				tail->next = fi;
-			tail = fi;
-
-			ret = fi_ibv_alloc_info(ctx_list[i], &fi,
-						&verbs_dgram_domain);
+		for (j = 0; j < VERBS_NUM_DOMAIN_TYPES; j++) {
+			ret = fi_ibv_alloc_info(ctx_list[i], &fi, ep_type[j]);
 			if (!ret) {
-				tail->next = fi;
+				if (!*all_infos)
+					*all_infos = fi;
+				else
+					tail->next = fi;
```

Comment:
Can above changes go in a separate patch?

Reply:
This was a simplification to facilitate the XRC update; it did not address a bug in the code directly. Going forward, I'll take more care where it makes sense to separate out this kind of simplification from the code that is being added.